ros_buildfarm/templates/doc/rosdoc2_job.xml.em

```diff
@@ -0,0 +1,247 @@
+        'echo "# BEGIN SECTION: Generate Dockerfile - doc task"',
+        'export TZ="%s"' % timezone,
+        'export PYTHONPATH=$WORKSPACE/ros_buildfarm:$PYTHONPATH',
+        'if [ "$force" = "true" ]; then FORCE_FLAG="--force"; fi',
+        'python3 -u $WORKSPACE/ros_buildfarm/scripts/doc/run_rosdoc2_job.py' +
+        ' ' + os_name +
+        ' ' + os_code_name +
+        ' ' + arch +
+        ' ' + ' '.join(repository_args) +
+        ' --dockerfile-dir $WORKSPACE/docker_generating_docker',
+        'echo "# END SECTION"',
+        '',
+        'echo "# BEGIN SECTION: Build Dockerfile - generating doc task"',
+        'cd $WORKSPACE/docker_generating_docker',
+        'python3 -u $WORKSPACE/ros_buildfarm/scripts/misc/docker_pull_baseimage.py',
+        'docker build --force-rm -t rosdoc2_task_generation.%s_%s .' % (rosdistro_name, doc_repo_spec.name.lower()),
+        'echo "# END SECTION"',
+        '',
+        'echo "# BEGIN SECTION: Run Dockerfile - generating doc task"',
+        'rm -fr $WORKSPACE/docker_doc',
+        'mkdir -p $WORKSPACE/docker_doc',
+        'docker run' +
+        ' --rm ' +
+        ' --cidfile=$WORKSPACE/docker_generating_docker/docker.cid' +
+        ' -e=HOME=/home/buildfarm' +
+        ' -v $WORKSPACE/ros_buildfarm:/tmp/ros_buildfarm:ro' +
+        ' -v $WORKSPACE/rosdoc2:/tmp/rosdoc2:ro' +
+        ' -v $WORKSPACE/ws:/tmp/ws' +
+        ' -v $WORKSPACE/generated_documentation:/tmp/generated_documentation' +
+        ' -v $WORKSPACE/docker_doc:/tmp/docker_doc' +
+        hgcache_mount_arg +
+        ' rosdoc2_task_generation.%s_%s' % (rosdistro_name, doc_repo_spec.name.lower()),
+        'echo "# END SECTION"',
+    ]),
+))@
+@(SNIPPET(
+    'builder_shell',
+    script='\n'.join([
+        'if [ ! -f "$WORKSPACE/docker_doc/Dockerfile" ]; then',
+        '  exit 0',
+        'fi',
+        '',
+        '# monitor all subprocesses and enforce termination',
+        'python3 -u $WORKSPACE/ros_buildfarm/scripts/subprocess_reaper.py $$ --cid-file $WORKSPACE/docker_doc/docker.cid > $WORKSPACE/docker_doc/subprocess_reaper.log 2>&1 &',
+        '# sleep to give python time to startup',
+        'sleep 1',
+        '',
+        'echo "# BEGIN SECTION: Build Dockerfile - doc"',
+        '# build and run build_and_install Dockerfile',
+        'cd $WORKSPACE/docker_doc',
+        'python3 -u $WORKSPACE/ros_buildfarm/scripts/misc/docker_pull_baseimage.py',
+        'docker build --force-rm -t rosdoc2.%s_%s .' % (rosdistro_name, doc_repo_spec.name.lower()),
+        'echo "# END SECTION"',
+        '',
+        'echo "# BEGIN SECTION: Run Dockerfile - doc"',
+        'docker run' +
+        ' --rm ' +
+        ' --cidfile=$WORKSPACE/docker_doc/docker.cid' +
+        ' -v $WORKSPACE/ros_buildfarm:/tmp/ros_buildfarm:ro' +
+        ' -v $WORKSPACE/rosdoc2:/tmp/rosdoc2:ro' +
+        ' -v $WORKSPACE/ws:/tmp/ws' +
+        ' -v $WORKSPACE/generated_documentation:/tmp/generated_documentation' +
```

Comment:
As far as I can tell from running this locally, this generated_documentation directory is entirely unused. Can we remove it entirely from the rosdoc2 templates and container mappings or is there some future use for it?

Reply:
Yeah, I think you are right that it is entirely unused.  Let me give it a quick whirl on the test buildfarm, but assuming it is unused I'll remove it completely.

Reply:
Yep, unused.  Removed in c144255.